CMakeLists.txt

```diff
@@ -1191,8 +1191,8 @@ endif ()
 # so that fortran name mangling is detected before writing H5pubconf.h
 #-----------------------------------------------------------------------------
 # Set default name mangling : overridden by Fortran detection in fortran dir
-set (H5_FC_FUNC  "H5_FC_FUNC(name,NAME) name ## _")
-set (H5_FC_FUNC_ "H5_FC_FUNC_(name,NAME) name ## _")
+set (H5_FC_FUNC  "H5_FC_FUNC(name,NAME) name ## _" CACHE INTERNAL "Fortran name mangling macro for C identifiers without underscores")
```

Comment:
This is a bit of a sledgehammer approach similar to #5627, but these variables have to be at this scope for H5pubconf.h and relying on `PARENT_SCOPE` is a bit fragile since the relevant include files could be moved even further down in scope in the future.